### DISCUSSION TITLE
error when files exist and action auto re-run

### DISCUSSION THREAD

**Discussion by vvssttkk** [stefanzweifel/git-auto-commit-action#280] [Help] [answered]

**Comment by stefanzweifel** [comment, marked as answer]
The problem is, that `actions/checkout` checked out the repo with the branch [`438-generate-picture-from-puml`](https://github.com/quantum-ods/qmlcourse/actions/runs/3864122481/jobs/6586729459#step:3:376).
Internally the action does a [`git checkout --branch {name}`](https://github.com/stefanzweifel/git-auto-commit-action/blob/master/entrypoint.sh#L99-L112) which then leads to the error you encountered.

You could add the following step **before** `holowinski/plantuml-github-action@main`, to checkout the repository to the right branch.

```yaml
- run: git checkout -b ${{ env.PUBLISH_BRANCH }}
```

Or add the `skip_checkout: true`-option to the git-auto-commit step.

```diff
  - name: push diagrams
    uses: stefanzweifel/git-auto-commit-action@v4.16.0
    with:
      commit_message: auto-generate `**.svg` for `**.plantuml`
      branch: ${{ env.PUBLISH_BRANCH }}
      # file_pattern: '**.svg'
      push_options: --dry-run  # '--force'
      add_options: '-f'
+     skip_dirty_check : true
```

**Reply by stefanzweifel** [reply]
Not sure, if you're using GitHub Actions here, but you could also use a dedicated Action for this:

- https://github.com/JamesIves/github-pages-deploy-action
- https://github.com/peaceiris/actions-gh-pages
